traversal/procedure/GraphProcedure.java

```diff
@@ -181,39 +184,39 @@ public Builder() {
             this.vertices = new HashMap<>();
         }
 
-        // TODO: remove when the query planner orders vertices natively
-        public GraphProcedure orderAndbuild() {
-            return new GraphProcedure(orderedVertices());
-        }
-
         public GraphProcedure build() {
-            return new GraphProcedure(vertices.values().stream().sorted(Comparator.comparing(ProcedureVertex::order))
+            return new GraphProcedure(vertices.values().stream().sorted(comparing(ProcedureVertex::order))
                     .toArray(ProcedureVertex[]::new));
         }
 
-        /**
-         * TODO: this is a compatibility layer to translate edge-ordered procedures into vertex-ordered procedures
-         * we need to remove it once the query planner natively orders vertices instead of edges
-         */
-        private ProcedureVertex<?, ?>[] orderedVertices() {
-            ProcedureVertex<?, ?>[] orderedVertices = new ProcedureVertex[vertices.size()];
-            List<ProcedureVertex<?, ?>> vertexList = iterate(vertices.values()).toList();
-            vertexList.sort(Comparator.comparing(v -> v.lastInEdge() == null ? 0 : v.lastInEdge().order()));
-            for (int i = 0; i < vertexList.size(); i++) {
-                orderedVertices[i] = vertexList.get(i);
-                orderedVertices[i].setOrder(i);
+        private void streamlineVertices() {
```

Comment:
so interestingly, the traversal engine will already be able to backtrack through the zigzag in an optimal way.
Additionally, this kind of "push towards DFS" operation feels like a planning task, rather than a procedure creation task - notice that because we have this, we still need the `setOrder()` method on the vertex, which ideally we'd get rid of and make the `ProcedureVertex` less stateful.

Given all that I'm inclined to remove it from here for now. We can then choose convert it into a planning task that will allow us to construct the streamlined procedure from the plan directly and delete the capability from the traversal engine instead if we deem it more correct.

Reply:
That wasn't the impression I got from the traversal engine, but you probably know better. Removed.